src/visitors/semantic/utils.rs

```diff
@@ -89,3 +89,13 @@ pub fn map_node_to_location(uri: String, node: Rc<Node>) -> Location {
         },
     }
 }
+
+pub fn follow_function_pipes(c: Box<CallExpr>) -> MonoType {
```

Comment:
change this to 
```rust
pub fn follow_function_pipes(c: &Box<CallExpr>) -> &MonoType {
    if let Some(p) = &c.pipe {
        if let Expression::Call(call) = p {
            return follow_function_pipes(&call);
        }
    }

    &c.typ
}
```

Reply:
Ownership with these visitors using walk in libflux can get tricky, prefer to clone than untangle ownership